### PR TITLE
B2BORG-76 - Validation if the user is already on MD

### PR DESCRIPTION
**Pull request by arturmagalhaesjr**
**What problem is this solving?**

Added a forgotten feature regarding the B2BORG-76 ticket. 
This feature provides a validation when the user is added checking if it exists on MD preventing duplicated entries.

It's linked on [https://b2borg--sandboxusdev.myvtex.com/](https://b2borg--sandboxusdev.myvtex.com/)

### CODE REVIEW

File: node/resolvers/Mutations/Users.ts [hunk 1]
```diff
@@ -123,6 +133,14 @@ export const addUser = async (_: any, params: any, ctx: Context) => {
   } = ctx
 
   try {
+    const userExists = await getUser({ masterdata, params })
+
+    if (userExists?.orgId !== params?.orgId) {
+      throw new Error(
+        `User with email ${params.email} and orgId ${params.orgId} already exists in another org.`
```
Comment (polishq) [inline node/resolvers/Mutations/Users.ts:140]: I think we should throw an error even if they already exist in the same org. Like, if you're trying to add a user as admin but they already exist in the same org as a buyer, the app should throw an error. The correct operation in that case would be to edit the existing user.

**Review by polishq — CHANGES_REQUESTED** [review]

**Review by arturmagalhaesjr — COMMENTED** [review]
done

File: node/resolvers/Mutations/Users.ts [hunk 2]
```diff
@@ -38,6 +38,25 @@ const addUserToMasterdata = async ({ masterdata, params }: any) => {
   return newUser.DocumentId
 }
 
+const getUser = async ({ masterdata, params }: any) => {
+  return masterdata
+    .searchDocuments({
+      dataEntity: 'CL',
```
Comment (polishq) [inline node/resolvers/Mutations/Users.ts:44]:
```suggestion
      dataEntity: config.name,
```
CL is the profile system; it's okay if a user already has an entry there. The app needs to check the storefront-permissions MD table instead.

**Review by polishq — CHANGES_REQUESTED** [review]

**Review by arturmagalhaesjr — COMMENTED** [review]
done

**Review by polishq — APPROVED** [review]
LGTM

**Pull request merged**